### PR TITLE
add label value validation

### PR DESCRIPTION
**Pull request by anvddriesch**
Towards: https://github.com/giantswarm/giantswarm/issues/15124

This is a copy of the rule a) from gatekeeper. If a user in a restricted group or with an admin username tries to delete or rename label values on the `cluster` resource other than versions, it will be rejected.

Because this introduces common validation, I have split the files in the `aws` package which causes a bit of a messy PR. Hopefully the comments help.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -61,6 +61,8 @@ Validating Webhook:
 
 - In a `AWSMachineDeployment` resource, it validates the Machine Deployment ID is matching against `MachineDeployment` resource.
 
+- In a `Cluster` resource, the non-version label values are not allowed to be deleted or renamed by whitelisted users and users in restricted groups.
```
Comment (anvddriesch) [inline README.md:64]:
```suggestion
- In a `Cluster` resource, the non-version label values are not allowed to be deleted or renamed by admin users and users in restricted groups. 
```

File: pkg/aws/common.go [hunk 2]
```diff
@@ -45,12 +44,34 @@ func ValidMasterReplicas() []int {
 	return []int{1, 3}
 }
 
+// ValidLabelAdmins returns the list of accounts used to manipulate labels
+func ValidLabelAdmins() []string {
+	return []string{
+		"system:serviceaccount:giantswarm:api",
```
Comment (anvddriesch) [inline pkg/aws/common.go:50]: This is taken from `gatekeeper`. The username that is used by the `api` for label manipulation is hardcoded.

File: pkg/aws/common_validator.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/dylanmei/iso8601"
+	"github.com/giantswarm/microerror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
```
Comment (anvddriesch) [inline pkg/aws/common_validator.go:11]: This is the label value validation function. It  is generic so we can also use it for other resources if we want to.

File: pkg/aws/common_validator_test.go [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			result := PauseTimeIsValid(tc.value)
+
+			if result != tc.valid {
+				t.Fatalf("%s -  expected '%t' got '%t'\n", tc.name, tc.valid, result)
+			}
+		})
+	}
+}
+
```
Comment (anvddriesch) [inline pkg/aws/common_validator_test.go:171]: This is the test for the new function

File: pkg/aws/common_fetcher_test.go [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+	"context"
+	"strconv"
+	"testing"
+
+	releasev1alpha1 "github.com/giantswarm/apiextensions/v3/pkg/apis/release/v1alpha1"
+	"github.com/giantswarm/micrologger/microloggertest"
+
+	"github.com/giantswarm/aws-admission-controller/v2/pkg/unittest"
+)
+
```
Comment (anvddriesch) [inline pkg/aws/common_fetcher_test.go:13]: tests for fetchers moved here

File: pkg/aws/common_handler.go [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+	releasev1alpha1 "github.com/giantswarm/apiextensions/v3/pkg/apis/release/v1alpha1"
+	"github.com/giantswarm/k8sclient/v5/pkg/k8sclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/aws-admission-controller/v2/pkg/label"
+	"github.com/giantswarm/aws-admission-controller/v2/pkg/mutator"
+)
+
```
Comment (anvddriesch) [inline pkg/aws/common_handler.go:20]: general handling (not specific to validation or mutation) moved here

File: pkg/aws/common_validator.go [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+				key,
+				value,
+				newLabels[key]),
+			)
+		}
+	}
+
+	return nil
+}
+
```
Comment (anvddriesch) [inline pkg/aws/common_validator.go:31]: other validation related functions moved here

File: pkg/aws/common_validator_test.go [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+	"context"
+	"strconv"
+	"testing"
+
+	"github.com/giantswarm/micrologger/microloggertest"
+
+	"github.com/giantswarm/aws-admission-controller/v2/pkg/label"
+	"github.com/giantswarm/aws-admission-controller/v2/pkg/unittest"
+)
+
```
Comment (anvddriesch) [inline pkg/aws/common_validator_test.go:13]: validation related tests moved here

**Review by xh3b4sd — APPROVED** [review]

**Pull request merged**